.github/workflows/build.yaml

```diff
@@ -13,6 +13,29 @@ permissions:
 run-name: Build from ${{ github.ref }}
 
 jobs:
+  formatting-tests:
+    name: Consistency and formatting
+    runs-on: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.9"
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Formatting tests
+        run: |
+          poetry install --with dev
+          poetry run black --check .
+          poetry run isort --check .
+          poetry run autoflake --check .
```

Comment:
tests here also need to use the same flags used in pre-commit config file and pyproject (especially line-length)